controller/controller.go

```diff
@@ -0,0 +1,12 @@
+package controller
+
+import (
+	"context"
+
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+// Controller is a reconciler that allows registration with a controller-runtime Manager
+type Controller interface {
```

Comment:
Is this ~= Runnable from controller-runtime?

Reply:
Or Registrable

Reply:
Runnable implements Start() so it's slightly different. Registerable feels more awkward and perhaps less direct for what this intended to be used for. We could enforce that the controller is also a Reconciler as well if we wanted to make it a more tightly defined concept, but I'm not fully convinced that's necessary here. 

I personally still like calling this Controller.

Reply:
I guess you register controllers with the controller managers. SGTM